Objects/typeobject.c

```diff
@@ -924,12 +924,6 @@ type_get_annotations(PyTypeObject *type, void *context)
             PyObject *annotations = _PyDict_GetItemId(kls->tp_dict, &PyId___annotations__);
             PyObject *co_annotations = _PyDict_GetItemId(kls->tp_dict, &PyId___co_annotations__);
             int co_annotations_is_set = co_annotations && (co_annotations != Py_None);
-            if (annotations && co_annotations_is_set) {
```

Comment:
This one and the next block I removed were causing problems because typing.py was looking at `types.FunctionType.__annotations__`. That's an attribute object, but of course FunctionType is a type, so it went through this descriptor. I think it's harmless to let this pass.

Reply:
It's harmless if `type` is types.FunctionType, but it would indicate a bug on any other object.   It's intended as a runtime assertion about invariants.  If the implementation is perfect, we don't need the check!  But I always worry, particularly about my own code.

Perhaps we could revive the assertion by adding "and neither is an instance of PyGetSetDescr_Type"?

Reply:
I think ModuleType would have the same problem (and can't check right now because my CPython checkout is compiling a different branch), and it's not a descriptor.

Reply:
I imagine it does.  It has the same check.